### PR TITLE
+slick #1333 add zipWith to DBIOAction

### PR DESCRIPTION
**Pull request by hepin1989**
refs https://github.com/slick/slick/issues/1333

I don't know why ``zip`` is implemented not via`` flatmap``?

### CODE REVIEW

File: slick/src/main/scala/slick/dbio/DBIOAction.scala [hunk 1]
```diff
@@ -63,6 +63,14 @@ sealed trait DBIOAction[+R, +S <: NoStream, -E <: Effect] extends Dumpable {
       (r(0).asInstanceOf[R], r(1).asInstanceOf[R2])
     } (DBIO.sameThreadExecutionContext)
 
+  /** Run another action after this action, if it completed successfully, and zip the result
+    * of both actions with a function `f`, then create a new DBIOAction holding this result,
+    * If either of the two actions fails, the resulting action also fails. */
+  def zipWith[R2, E2 <: Effect,R3](a: DBIOAction[R2, NoStream, E2])(f:(R,R2) =>R3): DBIOAction[R3, NoStream, E with E2] =
+    SequenceAction[Any, ArrayBuffer[Any], E with E2](Vector(this, a)).map { r =>
+      f(r(0).asInstanceOf[R], r(1).asInstanceOf[R2])
```
Comment (szeiger) [inline slick/src/main/scala/slick/dbio/DBIOAction.scala]: Don't run f on sameThreadExecutionContext. zipWith should take an implicit ExecutionContext, like all other methods that take a user-supplied function.

**Comment by szeiger** [comment]
Good question. The overridden version could still be fused, even if the base version was implemented via flatMap. AFAIR it used to be implemented via AndThenAction, before I replaced that by the more flexible SequenceAction. We also didn't have trampolining in 3.0. Back then FlatMapAction could blow the stack but AndThenAction wouldn't.

File: slick/src/main/scala/slick/dbio/DBIOAction.scala (hunk 1 above)
Comment (hepin1989) [inline slick/src/main/scala/slick/dbio/DBIOAction.scala]: will update then,thank you:)

**Comment by hepin1989** [comment]
@szeiger will you merge this?then I could do a rebase.

**Comment by szeiger** [comment]
Sorry for the delay, this LGTM. Thanks!

**Pull request merged**